### ISSUE TITLE
Expected behavior for "Content-Type" header on GET requests for restJson1

### ISSUE THREAD

**Issue by mcmasn-amzn** [awslabs/smithy#1450]
Original issue: https://github.com/awslabs/smithy-rs/issues/1845

What is the expected behavior of smithy clients/servers in the restJson1 protocol for the Content-Type header?

The protocol tests have [one example showing that](https://github.com/awslabs/smithy/blob/603c0f2de0dc54c7ed5cfa0d1e2f1c63d521eed2/smithy-aws-protocol-tests/model/restJson1/malformedRequests/malformed-content-type.smithy#L14-L29) a response of HTTP 415 should be given if Content-Type is included on a POST request without a body. 

But what about other HTTP verbs like GET, DELETE, or OPTIONS? Should it always be considered a malformed request if Content-Type is included when the request body is empty?

**Comment by kstich** [comment]
Yes, the request should be considered malformed if a `Content-Type` is present without a body. [The `restJson1`'s suite of tests for this are in `malformed-content-type.smithy`.](https://github.com/smithy-lang/smithy/blob/main/smithy-aws-protocol-tests/model/restJson1/malformedRequests/malformed-content-type.smithy)
